primitives/npos-elections/src/lib.rs

```diff
@@ -144,9 +145,84 @@ pub type VoteWeight = u64;
 /// A type in which performing operations on vote weights are safe.
 pub type ExtendedBalance = u128;
 
-/// The score of an assignment. This can be computed from the support map via
-/// [`EvaluateSupport::evaluate`].
-pub type ElectionScore = [ExtendedBalance; 3];
+/// The score of an election. This is the main measure of an election's quality.
+#[derive(Clone, Copy, PartialEq, Eq, Encode, Decode, MaxEncodedLen, TypeInfo, Debug, Default)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct ElectionScore {
+	/// The minimal winner, in terms of total backing stake.
+	///
+	/// This parameter should be maximized.
+	pub minimal_stake: ExtendedBalance,
+	/// The sum of the total backing of all winners.
+	///
+	/// This parameter should maximized
+	pub sum_stake: ExtendedBalance,
+	/// The sum squared of the total backing of all winners, aka. the variance.
+	///
+	/// Ths parameter should be minimized.
+	pub sum_stake_squared: ExtendedBalance,
+}
+
+impl ElectionScore {
+	fn iter(&self) -> impl Iterator<Item = ExtendedBalance> {
+		vec![self.minimal_stake, self.sum_stake, self.sum_stake_squared].into_iter()
+	}
+
+	/// Compares two sets of election scores based on desirability, returning true if `self` is
+	/// strictly `threshold` better than `other`.
+	///
+	/// Evaluation is done in a lexicographic manner, and if each element of `self` is `self *
+	/// epsilon` greater or less than `other`.
+	pub fn strict_threshold_better(&self, other: &Self, threshold: impl PerThing) -> bool {
```

Comment:
Because this is a Copy type, you don't really need to take references here, in fact it may even be better if you don't, since it saves a pointer indirection.

Reply:
https://github.com/paritytech/substrate/pull/10834/commits/24f1f6e5eb42b88eea8fc4375dea6730dab09067